node/hyperbahn/service_proxy.js

```diff
@@ -548,6 +539,25 @@ function destroy() {
     self.rateLimiter.destroy();
 };
 
+ServiceDispatchHandler.prototype.initCircuits =
+function initCircuits() {
+    var self = this;
+
+    self.circuits = new Circuits({
+        timeHeap: self.channel.timeHeap,
+        timers: self.channel.timers,
+        random: self.random,
+        egressNodes: self.egressNodes,
+        config: self.circuitsConfig
+    });
+
+    self.circuits.circuitStateChangeEvent.on(onCircuitStateChange);
+
+    function onCircuitStateChange(stateChange) {
```

Comment:
The closure is necessary to ensure that `this` within `self.onCircuitStateChange` is bound to `self` (the dispatch handler). The same can be accomplished with `self.onCircuitStateChange.bind(self)`, but is a known performance gotcha.